### PR TITLE
Propagate failure/preemptation and set timeout for multinode training

### PR DESCRIPTION
**Pull request by yizhongw**

### CODE REVIEW

File: configs/beaker_configs/default_finetune_lora_multinode.yaml [hunk 1]
```diff
@@ -5,6 +5,9 @@ tasks:
     replicas: 4
     leaderSelection: true
     hostNetworking: true
+    propagateFailure: true
+    propagatePreemption: true
+    synchronizedStartTimeout: 30m
```
Comment (vwxyzjn) [inline configs/beaker_configs/default_finetune_lora_multinode.yaml]: This seems like a very generous timeout. What about setting it to like 5 mins?

**Review by vwxyzjn — COMMENTED** [review]
Ah cool change! What does `propagateFailure` do?

**Comment by yizhongw** [comment]
> Ah cool change! What does `propagateFailure` do?

It will terminate all replicas if one of them fails. My understanding is this might be quicker than the distributed training system (e.g., deepspeed) finds no communication and terminates its processes, but I am not sure. There is some documentation for these parameters [here](https://beaker-docs.apps.allenai.org/experiments/distributed-training.html#batch-jobs) in beaker doc.

File: configs/beaker_configs/default_finetune_lora_multinode.yaml (hunk 1 above)
Comment (yizhongw) [reply configs/beaker_configs/default_finetune_lora_multinode.yaml]: 5 mins might be too short as one replica might be waiting the resources for get started. Let me set this to 15 mins (the recommended value in beaker doc).

**Review by vwxyzjn — APPROVED** [review]

**Pull request merged**